### ISSUE TITLE
tf and CUDA recognise GPU, yet ray local instances start without available gpus

### ISSUE THREAD

**Issue by psklavos1** [adap/flower#2502]
### Describe the bug

I am trying to run a simple simulation like the one described in the simulation tutorial by Javier. The problem is that Although 
print("Available GPUs:", tf.config.list_physical_devices("GPU"))
print("CUDA_VISIBLE_DEVICES",os.getenv("CUDA_VISIBLE_DEVICES"))
output: 
Available GPUs: [PhysicalDevice(name='/physical_device:GPU:0', device_type='GPU')],
CUDA_VISIBLE_DEVICES 0.
When Ray instantiates the different clients, they are assigned:
Ray initialized with resources: {'node:__internal_head__': 1.0, 'CPU': 12.0, 'node:172.27.192.62': 1.0, 'object_store_memory': 1721531596.0, 'memory': 3443063195.0}

This inevitably leads to the following error, when gpu resources are specified: The system (CPUs=12.0, GPUs=0) does not meet the criteria to host at least one client with resources: {'num_cpus': 2, 'num_gpus': 0.33}. Lowering the `client_resources` could help.

### Steps/Code to Reproduce

Needs the whole implementation so no code is passed

### Expected Results

The expected result would be to instantiate the clients with GPU: 1.0 

### Actual Results

No GPU filed is specified

**Comment by jafermarq** [comment]
Hi @psklavos1, If you try to run the code in [`examples/simulation-tensorflow`](https://github.com/adap/flower/tree/main/examples/simulation-tensorflow) do you experience the same problem?

**Comment by psklavos1** [comment]
Yes. The same problem persists. I don't want to trouble you with debugging.
I would just like any points to examine in order to resolve the issue.
Thank in advance for your help

**Comment by jafermarq** [comment]
hmmm... did you add those `print("Available GPUs:", tf.config.list_physical_devices("GPU"))` in the same python file you later run `start_simulation`? Are you on Windows? (if yes, could you try WSL?)

**Comment by psklavos1** [comment]
Yes, they are at the same file(it is in the main.py). Essentialy, the code
is an effort to emulate your videos on flower simulation, using tensor
flow. I am already on Wsl.

**Comment by psklavos1** [comment]
Is there any Idea of what to do to fix the problem. My Gpu is fully recognised when executing: 
print("Available GPUs:", tf.config.list_physical_devices("GPU"))

and when trying to list:
from tensorflow.python.client import device_lib 
print(device_lib.list_local_devices())

But Ray still does not seem to initialize gpu device. Can I manually initialize to see the gpu in some way?

**Comment by jafermarq** [comment]
@alessiomora, do you know what could be causing this? @psklavos1 is trying to do simulation with TF in Window's WSL but his GPU isn't detected properly

**Comment by psklavos1** [comment]
Yes. I can verify the issue is the same with: 'num_gpus': 1.0

**Comment by Dknight27** [comment]
I had the same problem, but I was using pytorch.
`torch.cuda.is_available(): True
INFO flwr 2023-12-24 03:37:57,013 | app.py:178 | Starting Flower simulation, config: ServerConfig(num_rounds=10, round_timeout=None)
2023-12-24 03:37:59,156 INFO worker.py:1553 -- Started a local Ray instance.
INFO flwr 2023-12-24 03:38:01,192 | app.py:213 | Flower VCE: Ray initialized with resources: {'object_store_memory': 35014316851.0, 'memory': 71700072653.0, 'CPU': 40.0, 'node:10.139.50.89': 1.0}
INFO flwr 2023-12-24 03:38:01,192 | app.py:219 | Optimize your simulation with Flower VCE: https://flower.dev/docs/framework/how-to-run-simulations.html
INFO flwr 2023-12-24 03:38:01,192 | app.py:242 | Flower VCE: Resources for each Virtual Client: {'num_cpus': 1, 'num_gpus': 1.0}
WARNING flwr 2023-12-24 03:38:01,193 | ray_actor.py:144 | The ActorPool is empty. The system (CPUs=40.0, GPUs=0) does not meet the criteria to host at least one client with resources: {'num_cpus': 1, 'num_gpus': 1.0}. Lowering the `client_resources` could help.`

**Comment by WilliamLindskog** [comment]
Hi @psklavos1 

Thanks for raising this. Are you still experiencing this issue?

**Comment by WilliamLindskog** [comment]
Just tried with new versions of Flower and it seems that this problem has been resolved in newer versions. There are also new instructions on how to run simulations, specifically for TF for which one needs to [enable memory growth](https://www.tensorflow.org/guide/gpu#limiting_gpu_memory_growth) so multiple `ClientApp` instances can share GPU. 

Closing this issue for now.
